core/embed/trezorhal/stm32f4/i2c.c

```diff
@@ -171,9 +171,17 @@ HAL_StatusTypeDef i2c_transmit(uint16_t idx, uint8_t addr, uint8_t *data,
                                uint16_t len, uint32_t timeout) {
   return HAL_I2C_Master_Transmit(&i2c_handle[idx], addr, data, len, timeout);
 }
+
 HAL_StatusTypeDef i2c_receive(uint16_t idx, uint8_t addr, uint8_t *data,
                               uint16_t len, uint32_t timeout) {
-  return HAL_I2C_Master_Receive(&i2c_handle[idx], addr, data, len, timeout);
+  HAL_StatusTypeDef ret =
+      HAL_I2C_Master_Receive(&i2c_handle[idx], addr, data, len, timeout);
+#ifdef USE_OPTIGA
+  if (idx == OPTIGA_I2C_INSTANCE) {
```

Comment:
oh, wait, `idx` is not a _device_ on a bus, but a _bus_ out of theoretical multiple ones, right?
please add a comment mentioning this

Reply:
https://github.com/trezor/trezor-firmware/pull/3303/commits/52ce05eecebe1b1c6872e051008bd65b4ada94ab